src/scripts/main.js

```diff
@@ -1,3 +1,193 @@
-'use strict';
+const table = document.querySelector('table');
+const headers = document.querySelectorAll('th');
+const tbody = document.querySelector('tbody');
+const rows = [...tbody.rows];
 
-// write code here
+document.addEventListener('DOMContentLoaded', () => {
+  headers.forEach((header, index) => {
+    let sortDirection = 'asc';
+    let activeHeader = null;
+
+    header.addEventListener('click', () => {
+      if (activeHeader !== header) {
+        sortDirection = 'asc';
+        activeHeader = header;
+      } else {
+        sortDirection = sortDirection === 'asc' ? 'desc' : 'asc';
+      }
+
+      sortTable(table, index, sortDirection);
+    });
+  });
+});
+
+function sortTable(tableT, columnIndex, direction) {
+  rows.sort((rowA, rowB) => {
+    const cellA = rowA.cells[columnIndex].textContent.trim().toLowerCase();
+    const cellB = rowB.cells[columnIndex].textContent.trim().toLowerCase();
+
+    let valueA, valueB;
+
+    if (columnIndex === 3 || columnIndex === 4) {
+      valueA = parseFloat(cellA.replace(/[$, ]/g, ''));
+      valueB = parseFloat(cellB.replace(/[$, ]/g, ''));
```

Comment:
The code assumes that columns with index 3 and 4 contain numeric values formatted as currency. Ensure that these indices are correct for your table structure. If the table structure changes, this logic might need updating.